test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs

```diff
@@ -586,6 +585,80 @@ public void InitializeXPlatCodeCoverageShouldAddXPlatInProcProcDataCollectoPrope
+            this.settingsProvider.SetActiveRunSettings(runsettings);
+
+            this.executor.Initialize("MyDataCollector;SomeSetting=SomeValue;AnotherSetting=AnotherValue");
+
+            Assert.IsNotNull(this.settingsProvider.ActiveRunSettings);
+            Assert.AreEqual(string.Join(Environment.NewLine,
+                "<?xml version=\"1.0\" encoding=\"utf-16\"?>",
+                "<RunSettings>",
+                "  <RunConfiguration>",
+                "    <TestAdaptersPaths>c:\\AdapterFolderPath</TestAdaptersPaths>",
+                "  </RunConfiguration>",
+                "  <DataCollectionRunSettings>",
+                "    <DataCollectors>",
+                "      <DataCollector friendlyName=\"MyDataCollector\" enabled=\"True\">",
+                "        <Configuration>",
+                "          <SomeSetting>SomeValue</SomeSetting>",
+                "          <AnotherSetting>AnotherValue</AnotherSetting>",
+                "        </Configuration>",
+                "      </DataCollector>",
+                "    </DataCollectors>",
+                "  </DataCollectionRunSettings>",
+                "</RunSettings>"), this.settingsProvider.ActiveRunSettings.SettingsXml);
+        }
+
+        [TestMethod]
+        public void InitializeShouldCreateConfigurationsForExistingDataCollectorInRunSettings()
+        {
+            var runsettingsString = string.Format(DefaultRunSettings,
+                "<DataCollector friendlyName=\"MyDataCollector\" enabled=\"False\">" +
+                "  <Configuration>" +
+                "    <SomeSetting>SomeValue</SomeSetting>" +
+                "  </Configuration>" +
+                "</DataCollector>");
+            var runsettings = new RunSettings();
+            runsettings.LoadSettingsXml(runsettingsString);
+            this.settingsProvider.SetActiveRunSettings(runsettings);
+
+            this.executor.Initialize("MyDataCollector;AnotherSetting=AnotherValue");
+
+            Assert.AreEqual(string.Join(Environment.NewLine,
+                "<?xml version=\"1.0\" encoding=\"utf-16\"?>",
+                "<RunSettings>",
+                "  <RunConfiguration>",
+                "    <TestAdaptersPaths>c:\\AdapterFolderPath</TestAdaptersPaths>",
+                "  </RunConfiguration>",
+                "  <DataCollectionRunSettings>",
+                "    <DataCollectors>",
+                "      <DataCollector friendlyName=\"MyDataCollector\" enabled=\"True\">",
+                "        <Configuration>",
+                "          <SomeSetting>SomeValue</SomeSetting>",
+                "          <AnotherSetting>AnotherValue</AnotherSetting>",
+                "        </Configuration>",
+                "      </DataCollector>",
+                "    </DataCollectors>",
+                "  </DataCollectionRunSettings>",
+                "</RunSettings>"), this.settingsProvider.ActiveRunSettings.SettingsXml);
```

Comment:
Add also some testing in case you have 2 collectors enabled:

--collect "datacollector1;arg1=val1,arg2=val2;datacollector2;arg3=val3,arg4=val4"

Reply:
In previous implementation `dotnet test --collect "Code Coverage;XPlat Code Coverage"` doesn't add two collectors. It will add only 1 with friendly name "Code Coverage;XPlat Code Coverage"
```
    <DataCollectors>
      <DataCollector friendlyName="MyDataCollector;AnotherCollector" enabled="True" />
    </DataCollectors>
```

Reply:
You are basing it probably on unit tests results but this class that you are testing is getting string which was already splitted by ";". This is why you need to add acceptance tests :). Another thing that there is already special processor for "Code Coverage".

Check this:
```
D:\examples\xunit170> dotnet test --collect "Code Coverage;XPlat Code Coverage"
  Determining projects to restore...
  All projects are up-to-date for restore.
  xunit170 -> D:\examples\xunit170\bin\Debug\net6.0\xunit170.dll
Test run for D:\examples\xunit170\bin\Debug\net6.0\xunit170.dll (.NETCoreApp,Version=v6.0)
Microsoft (R) Test Execution Command Line Tool Version 17.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Attachments:
  D:\examples\xunit170\TestResults\07f5cdaa-e532-43a0-a96a-4533eba6df87\coverage.cobertura.xml
  D:\examples\xunit170\TestResults\07f5cdaa-e532-43a0-a96a-4533eba6df87\UseThisAccount_JACHOCHO-5820_2021-11-11.12_22_46.coverage
Passed!  - Failed:     0, Passed:     1, Skipped:     0, Total:     1, Duration: < 1 ms - xunit170.dll (net6.0)
```

Reply:
From code, we can add multiple `collect` commands. So to specify multiple collectors we can use this.
`dotnet text --collect:"Collector1" --collect:"Collector2"`
With configurations.
`dotnet text --collect:"Collector1;arg1=val1" --collect:"Collector2;arg2=val2"`

Reply:
Acceptance tests accepts runsettings string and doesn't take command line arguments. @nohwnd Do you know if there are any acceptance tests which takes both runsettings and command line arguments?

Reply:
what if someone is already using --collect "datacollector1;datacollector2"? Your code would throw exception. I think we shouldn't make such breaking change

Reply:
@jakubch1 We did not find such usage on github, and we can't say for sure, but I agree to throw error in that case. Logger parameter does the same. 

@fhnaseer ExecuteTestsWithDataCollection seems to be providing both runsettings as well as command line parameters. or is that not what you want?  

https://github.com/microsoft/vstest/blob/ad32654bc155a059fda263f0ec99b445058d84ea/test/Microsoft.TestPlatform.AcceptanceTests/DataCollectionTests.cs#L36-L37

Reply:
Added acceptance tests in test/Microsoft.TestPlatform.AcceptanceTests/CodeCoverageTests.cs